app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -113,6 +113,7 @@ def alternate_machine_id(machine_id)
         # providers store it in downcase, so, we let the upcase/downcase logic to other methods with more
         # business knowledge.
         # 1 - https://bugzilla.redhat.com/show_bug.cgi?id=1294461
+        return nil if machine_id.nil? || machine_id.length != 32 || machine_id[/\H/]
```

Comment:
it looks like there is already a check for machine_id.nil on line 108 ? and is the big explanation on lines 109 to 115 still stand?

Reply:
Oh! I didn't see that line. Thanks!
And I think that explanation still apply. We cannot be sure if the kernel of the machine is already patched. I think we should keep it just in case.

Reply:
I think this would be better:

```ruby
return nil if machine_id.to_s.length != 32 || machine_id.to_s[/\H/]
```
Also, what does `\H` matches? That's a new one to me.

Reply:
From https://ruby-doc.org/core-2.2.0/Regexp.html:
> `/\H/` - A non-hexdigit character ([^0-9a-fA-F])

:)

Reply:
Oh, nice!